iati/validator.py

```diff
@@ -272,6 +272,69 @@ def get_warnings(self):
         return [err for err in self if err.status == 'warning']
 
 
+def _extract_codes_from_attrib(dataset, split_xpath, condition=None):
+    """Extract codes for checking from a Dataset. The codes are being extracted from attributes.
+
+    Args:
+        dataset (iati.data.Dataset): The Dataset to check Codelist values within.
+        split_xpath (list of str): The sections of the XPath, split by `/`.
+        condition (str): An optional XPath expression to limit the scope of what is extracted.
+
+    Returns:
+        list of tuple: A tuple in the format: `(str, int)` - The `str` is a matching code from within the Dataset; The `int` is the sourceline at which the parent element is located.
+
+    """
+    parent_el_xpath = '/'.join(split_xpath[:-1])
+    last_xpath_section = split_xpath[-1:][0]
```

Comment:
I think this functionally equivalent to:

```python
last_xpath_section = split_xpath[-1]
```

Reply:
I.e. the code that’s there makes a list of all items from the last one onwards, then takes the first item of that list.

Reply:
So it is!

Reply:
d12d03fc196406326b43fd59f1c1fc366d558b4b